### PR TITLE
Adding NXP NavQPlus to detected boards

### PR DESCRIPTION
**Pull request by steamfoundry**
dding the NXP NavQPlus board:
Board documentation available [here:](https://nxp.gitbook.io/8mpnavq/)
(Based on the the NXP i.MX 8M Plus SOC so the chip is already detected)

Future boards using the same SOC may need to edit the board name or detection line in bin/detect.py so it doesn't list it as an NXP NavQ based board

Tested on Ubuntu 22.04 built on top of NXP Yocto 5.15

Added for the Hovergames 3
Team McFly
2023
Instructions followed : https://learn.adafruit.com/adding-a-single-board-computer-to-platformdetect-for-blinka/overview

### CODE REVIEW

**Comment by steamfoundry** [comment]
@makermelissa Can you review this when you have time please?

File: adafruit_platformdetect/board.py [hunk 1]
```diff
@@ -418,6 +418,8 @@ def _imx8mx_id(self) -> Optional[str]:
             return boards.MAAXBOARD
         if "Phanbell" in board_value:
             return boards.CORAL_EDGE_TPU_DEV
+        if "NXP i.MX8MPlus SOM " in board_value:
```
Comment (makermelissa) [inline adafruit_platformdetect/board.py:421]: Is the trailing space in the name required or something you accidentally added?

Comment (steamfoundry) [reply adafruit_platformdetect/board.py:421]: Purely typo. Thanks for catching it

Comment (makermelissa) [inline adafruit_platformdetect/board.py:421]:
```suggestion
        if "NXP i.MX8MPlus SOM" in board_value:
```

**Comment by makermelissa** [comment]
Can you run pre-commit so that it fixes the formatting and the checks pass? Otherwise, the code itself looks good.

**Comment by steamfoundry** [comment]
Sorry about that @makermelissa.  I started looking at the Blinka code and found I need to detect this board separately.  Otherwise some of the GPIO and UART numbers won't line up and I didn't want to risk breaking existing code.


I didn't realize Github auto-adds to the Pull Request when I pushed in the new commits.
Here is the results from running the test script.

```
user@imx8mpnavq:~/src/Adafruit_Python_PlatformDetect/bin$ ./detect.py 
Board Detection Test

Check that the Chip and Board IDs match your board and that this it is
correctly detecting whether or not it is a Linux board.

Chip id:  IMX8MP
Board id:  NXP_IMX8MPLUS_SOM

Linux Detection
---------------
Is this an embedded Linux system? True

Raspberry Pi Boards
-------------------
Is this a Pi 3B+? False
Is this a Pi 4B? False
Is this a 40-pin Raspberry Pi? False
Is this a Raspberry Pi Compute Module? False

Other Boards
-------------------
Is this a Siemens Simatic IOT2000 Gateway? False
Is this a 96boards board? False
Is this a BeagleBone board? False
Is this a Giant board? False
Is this a Coral Dev board? False
Is this a MaaXBoard? False
Is this a SiFive board?  False
Is this a PYNQ board? False
Is this a Rock Pi board? False
Is this a NanoPi board? False
Is this a Khadas VIM3 board? False
Is this a Clockwork Pi board? False
Is this a Seeed Board? False
Is this a UDOO board? False
Is this an ASUS Tinker board? False
Is this an STM32MP1 board? False
Is this a generic Linux PC? False
Is this an OS environment variable special case? False
NXP NavQ board detected.

```

**Comment by makermelissa** [comment]
Awesome. Did you want me to wait before merging this?

**Comment by steamfoundry** [comment]
No, if you don't see any issues it can be merged.
Thank you @makermelissa

**Review by makermelissa — APPROVED** [review]

**Comment by steamfoundry** [comment]
Thank you!

**Pull request merged**